adobecom/milo
# [MWPW-153611] [Gray Box] environment aware links

## Description
Links on Graybox pages need to be environment aware.

**Examples of desired links conversions:** 
When on _**https://pocone.graybox.adobe.com/some-path**_:
`https://www.adobe.com/` -> `https://pocone.graybox.adobe.com/`
When on **_https://max24.business-graybox.adobe.com/some-path_**
`https://business.adobe.com/` -> `https://max24.business-graybox.adobe.com/` 

where 
`pocone` / `max` are graybox environment names. These can be anything.
`graybox` / `business-graybox` are the names of the graybox environments sharepoint container folder. Each supported Milo consumer folder will have a `–graybox` variant.

## Related Issue
Resolves: [MWPW-153611](https://jira.corp.adobe.com/browse/MWPW-153611)

## Testing instructions
`1.` go to the test URL
`2.` using the developer tools, add a breakpoint in `scripts.js`, where `LIBS` is defined.

<img width="1549" alt="Screenshot 2024-07-24 at 20 37 23" src="https://github.com/user-attachments/assets/4a3898ef-d82c-4037-8b94-e0582e5d1207">

`3.` refresh the page, when the execution hits the break point. Overwrite, using the console, the `setLibs` function with `() => 'https://mwpw-153611-convert-graybox-urls--milo--robert-bogos.hlx.live/libs'`

![image](https://github.com/user-attachments/assets/dc0293cf-d706-46b2-84c9-7fc449690cde)

`4.` resume the execution
`5.` scroll down on the page and inspect the `Link to be converted` link. It should send to `https://pocone.business-graybox.adobe.com/...`

<img width="765" alt="Screenshot 2024-07-24 at 20 45 27" src="https://github.com/user-attachments/assets/09af4fd8-a1c4-4b6c-ab76-a587dbab1eb7">

if you don't overwrite the `LIBS`, the `Link to be converted` link will send to `https://business.adobe.com/...`

## Test URL
https://pocone.business-graybox.adobe.com/customer-success-stories/graybox-links-conversion.html

## URL for PSI check: (irrelevant for testing the changes)
https://mwpw-153611-convert-graybox-urls--milo--robert-bogos.hlx.page/drafts/rbogos/page-default?martech=off
